spec.bs

```diff
@@ -3309,6 +3312,74 @@ left over to be leaked to the next {{Document}}.
      mind, and could evolve in ways that give this specification unwanted side-effects.
 </div>
 
+<h4 id=unfenced-top-navigation-target>The "<code>_unfencedTop</code>" navigation target</h4>
+
+Fenced frames use an additional reserved navigation target "<code>_unfencedTop</code>" to navigate.
+This section patches the relevant portions of the [[HTML]] spec to add <code>_unfencedTop</code>.
+
+<div algorithm="unfenced-top-keyword">
+  Modify [=valid navigable target name or keyword=] to include a new keyword named <code>
+  _unfencedTop</code>.
+
+  Modify the below paragraph to replace the following text:
+
+    * "top" means the [=top-level traversable=] of the [=navigable=] that the link or script is in,
+
+  with: 
+
+    * "top" means the [=navigable/traversable navigable=] of the [=navigable=] that the link or
+      script is in, "outermost top" means the [=top-level traversable=] of the [=navigable=] that
+      the link or script is in, 
+
+  Note: This change is necessary because this spec adds a distinction between
+  [=navigable/traversable navigable=] and [=top-level traversable=], which were previously
+  functionally equivalent.
+
+  In the table below, add a column named "Effect in a fenced frame". The value for all existing
+  rows in this column is the same as "ordinary effect". Then, add two rows which read:
+
+  <table>
+    <thead>
+      <tr>
+        <th rowspan=2>Keyword</th>
+        <th rowspan=2>Ordinary effect</th>
+        <th colspan=2>Effect in an <code>iframe</code> with...</th>
+        <th rowspan=2>Effect in a fenced frame</th>
+      </tr>
+      <tr>          
+        <th><code data-x="">sandbox=""</code></th>
+        <th><code data-x="">sandbox="allow-top-navigation"</code></th>
+      </tr>
+    </thead>
+
+    <tbody>
+      <tr>
+        <td><code>_unfencedTop</code> if outermost top is current</td>
+        <td>current</td>
+        <td>current</td>
+        <td>current</td>
+        <td>current</td>
+      </tr>
+      <tr>
+        <td><code>_unfencedTop</code> if outermost top is not current</td>
+        <td>outermost top</td>
+        <td>none</td>
+        <td>outermost top</td>
+        <td>outermost top</td>
+      </tr>
+    </tbody>
+  </table>
+</div>
+
+<div algorithm="unfenced-top-rules">
+  Add an extra step to [=the rules for choosing a navigable=] between the current steps 6 and 7, 
+  that reads:
+
+  6.5. Otherwise, if name is an ASCII case-insensitive match for "_unfencedTop" and
+  |currentNavigable|'s [=navigable/traversable navigable=] is a [=fenced navigable container/fenced
+  navigable=], set <var ignore=''>chosen</var> to |currentNavigable|'s [=top-level traversable=].
```

Comment:
What if currentNavigable's traversable navigable is not a fenced navigable? Don't we break the case where you might use `_unfencedTop` in a top-level frame, or even an iframe, which should work, per the table above?

Reply:
The table is not currently correct, see Liam's comment thread above. Outside of a fenced frame, `_unfencedTop` should just open a new context, same as a ["name that doesn't exist"](https://html.spec.whatwg.org/multipage/document-sequences.html#valid-navigable-target-name-or-keyword:~:text=name%20that%20doesn%27t%20exist)

So I think it's fine for the algorithm to continue if this condition is not met, at which point the steps for requesting a new top-level traversable will occur. I fixed the table in the latest patch.

Reply:
Also: this isn't 100% how the Chromium implementation works right now, because only API-generated fenced frames can use `unfencedTop.`  I'm fixing that in [this CL](https://crrev.com/c/6017449), so that all fenced frames can use the new keyword, and then the spec and Chromium code will match.